src/library/dao/trackdao.cpp

```diff
@@ -1523,14 +1523,30 @@ TrackPointer TrackDAO::getTrackById(TrackId trackId) const {
             updateTrackFromSourceMode =
                     SoundSourceProxy::UpdateTrackFromSourceMode::Newer;
         }
-        SoundSourceProxy(pTrack).updateTrackFromSource(
-                updateTrackFromSourceMode,
-                SyncTrackMetadataParams::readFromUserSettings(*m_pConfig));
-        if (kLogger.debugEnabled() && pTrack->isDirty()) {
-            kLogger.debug()
-                    << "Updated track metadata from file tags:"
-                    << pTrack->getFileInfo().lastModified()
-                    << pTrack->getMetadata();
+        DEBUG_ASSERT(!pTrack->isDirty());
```

Comment:
I have added lots of debug assertions here to verify the assumptions.

Reply:
It should, since you just set pTrack->markClean(); above.